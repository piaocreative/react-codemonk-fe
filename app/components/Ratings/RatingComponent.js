import React, { useState } from 'react';
import Rating from 'react-rating';
import PropTypes from 'prop-types';
import Badge from '../Badge';
import P from '../P';
import { RatingWrapper, EmptyIcon, FullIcon } from './rating-styles';
export const RatingComponent = props => {
  const { index, ratingItem, onChangeRating, itemList, setError, forceValidate } = props;
  const { label, rating } = ratingItem;
  const [rateLabel, setRateLabel] = useState(rating === undefined ? '1-10' : rating);
  const [ratingHoverIndex, setRatingHoverIndex] = useState(0);
  const [ratingItemCurrent, setRatingItemCurrent] = useState(ratingItem);
  const onChangeHandler = (indexRating, labelRating, value) => {
    onChangeRating(indexRating, value, labelRating);
    setRatingItemCurrent({ label, rating: value });
  };

  return (
    <RatingWrapper index={index}>
      <div className="rating-item">
        <Badge className="title badge-md secondary">{label}</Badge>
        <div className="ratings">
          <div className="rating-bars">
            <Rating
              data-testid="Rating"
              quiet
              onHover={value => setRatingHoverIndex(value)}
              onChange={value => {
                if (itemList.filter(item => item.rating === value).length === 2 && forceValidate) {
                  setError('Please do not rate any more than two skills at the same level.');
                } else {
                  setError('');
                  setRateLabel(value);
                  onChangeHandler(index, label, value);
                }
              }}
              start={0}
              initialRating={ratingItemCurrent.rating}
              fullSymbol={Array(10)
                .fill(0)
                .map(() => (
                  <FullIcon />
                ))}
              emptySymbol={Array(10)
                .fill(0)
                .map((_, index1) => (
                  <EmptyIcon icon={ratingHoverIndex} index={index1} />
                ))}
              stop={10}
            />
          </div>
          <P className="p16 rating-label m-0" id={label}>
            {rateLabel}
          </P>
        </div>
      </div>
    </RatingWrapper>
  );
};
RatingComponent.propTypes = {
  index: PropTypes.number,
  ratingItem: PropTypes.object,
  onChangeRating: PropTypes.func,
  forceValidate: PropTypes.bool,
  setError: PropTypes.func,
  itemList: PropTypes.array,
};
RatingComponent.defaultProps = {
  index: 0,
  ratingItem: {},
  onChangeRating: () => {},
};
export default RatingComponent;
