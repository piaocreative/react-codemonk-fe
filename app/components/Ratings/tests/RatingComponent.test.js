import React from 'react';
import renderer from 'react-test-renderer';
import 'whatwg-fetch';
import { shallow } from 'enzyme';
import { RatingComponent as MainForm } from '../RatingComponent';

jest.mock('utils/request');

const props = {
  index: 0,
  ratingItem: {
    label: 'test',
    rating: 4,
  },
  onChangeRating: jest.fn(),
};

const getWrapper = () => shallow(<MainForm {...props} />);

describe('<RatingComponent />', () => {
  it('should render and match the snapshot', () => {
    const renderedComponent = renderer.create(<MainForm />).toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('test functions', () => {
  // test('test onHover', () => {
  //   const test = getWrapper().find('[data-testid="Rating"]');
  //   test.props().onHover('test');
  //   test.props().onChange();
  // });
  test('test onHover', () => {
    props.ratingItem = {};
    const test = getWrapper().find('[data-testid="Rating"]');
    test.props().onHover();
  });
});
