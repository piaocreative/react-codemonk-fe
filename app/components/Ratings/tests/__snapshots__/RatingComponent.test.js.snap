// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingComponent /> should render and match the snapshot 1`] = `
<div
  className="rating-styles__RatingWrapper-yq5jrq-0 wKcYD"
>
  <div
    className="rating-item"
  >
    <span
      className="Badge-n20gz6-0 euqJll title badge-md secondary"
    />
    <div
      className="ratings"
    >
      <div
        className="rating-bars"
      >
        <span
          onMouseLeave={[Function]}
          style={
            Object {
              "direction": "ltr",
              "display": "inline-block",
            }
          }
        >
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={Object {}}
            >
              <div
                className="rating-styles__EmptyIcon-yq5jrq-1 jJTvVo"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "0%",
                }
              }
            >
              <div
                className="rating-styles__FullIcon-yq5jrq-2 folHcU"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={Object {}}
            >
              <div
                className="rating-styles__EmptyIcon-yq5jrq-1 jJTvVo"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "0%",
                }
              }
            >
              <div
                className="rating-styles__FullIcon-yq5jrq-2 folHcU"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={Object {}}
            >
              <div
                className="rating-styles__EmptyIcon-yq5jrq-1 jJTvVo"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "0%",
                }
              }
            >
              <div
                className="rating-styles__FullIcon-yq5jrq-2 folHcU"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={Object {}}
            >
              <div
                className="rating-styles__EmptyIcon-yq5jrq-1 jJTvVo"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "0%",
                }
              }
            >
              <div
                className="rating-styles__FullIcon-yq5jrq-2 folHcU"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={Object {}}
            >
              <div
                className="rating-styles__EmptyIcon-yq5jrq-1 jJTvVo"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "0%",
                }
              }
            >
              <div
                className="rating-styles__FullIcon-yq5jrq-2 folHcU"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={Object {}}
            >
              <div
                className="rating-styles__EmptyIcon-yq5jrq-1 jJTvVo"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "0%",
                }
              }
            >
              <div
                className="rating-styles__FullIcon-yq5jrq-2 folHcU"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={Object {}}
            >
              <div
                className="rating-styles__EmptyIcon-yq5jrq-1 jJTvVo"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "0%",
                }
              }
            >
              <div
                className="rating-styles__FullIcon-yq5jrq-2 folHcU"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={Object {}}
            >
              <div
                className="rating-styles__EmptyIcon-yq5jrq-1 jJTvVo"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "0%",
                }
              }
            >
              <div
                className="rating-styles__FullIcon-yq5jrq-2 folHcU"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={Object {}}
            >
              <div
                className="rating-styles__EmptyIcon-yq5jrq-1 jJTvVo"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "0%",
                }
              }
            >
              <div
                className="rating-styles__FullIcon-yq5jrq-2 folHcU"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={Object {}}
            >
              <div
                className="rating-styles__EmptyIcon-yq5jrq-1 jJTvVo"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "0%",
                }
              }
            >
              <div
                className="rating-styles__FullIcon-yq5jrq-2 folHcU"
              />
            </span>
          </span>
        </span>
      </div>
      <p
        className="P-tiusvc-0 hxrvHG p16 rating-label m-0"
      >
        1-10
      </p>
    </div>
  </div>
</div>
`;
